tests/admin/test_controller.py

```diff
@@ -1081,6 +1112,236 @@ def test_classifications(self):
                           self.manager.admin_work_controller.classifications,
                           lp.identifier.type, lp.identifier.identifier)
 
+    def test_validate_cover_image(self):
+        base_path = os.path.split(__file__)[0]
+        resource_path = os.path.join(base_path, "..", "files", "images")
+
+        path = os.path.join(resource_path, "blue_small.jpg")
+        too_small = Image.open(path)
+
+        result = self.manager.admin_work_controller._validate_cover_image(too_small)
+        eq_(INVALID_COVER_IMAGE.uri, result.uri)
+        eq_("Cover image must be at least 600px in width and 900px in height.", result.detail)
+
+        path = os.path.join(resource_path, "blue.jpg")
+        valid = Image.open(path)
+        result = self.manager.admin_work_controller._validate_cover_image(valid)
+        eq_(True, result)
+
+    def test_process_cover_image(self):
+        work = self._work(with_license_pool=True, title="Title", authors="Authpr")
+
+        base_path = os.path.split(__file__)[0]
+        resource_path = os.path.join(base_path, "..", "files", "images")
+        path = os.path.join(resource_path, "blue.jpg")
+        original = Image.open(path)
+        processed = Image.open(path)
+
+        # Without a title position, the image won't be changed.
+        self.manager.admin_work_controller._process_cover_image(work, processed, "none")
+
+        image_histogram = original.histogram()
+        expected_histogram = processed.histogram()
+
+        root_mean_square = math.sqrt(reduce(operator.add,
+                                            map(lambda a,b: (a-b)**2, image_histogram, expected_histogram))/len(image_histogram))
+        assert root_mean_square < 10
```

Comment:
This is validating that the processed image is not changed that much from the original?

Reply:
Yeah, you're comparing the actual to expected with some fuzz accounting for slight variations in how PIL decides to handle the image.

Reply:
Correct.